### ISSUE TITLE
No indentation after operator

### ISSUE THREAD

**Issue by dhoepelman** [scala-ide/scalariform#216]
Akka HTTP uses a routing DSL in which routes are constructed as follows:
```
def routeCombined =
  getSomething { somevar =>
    route1 ~
    route2 ~
    route3 
  }
```
Scalariform formats this as:
```
def routeCombined =
  getSomething { somevar =>
    route1 ~
      route2 ~
      route3 
  }
```

This causes unnecessary indenting and less readability. [Here's a larger example](https://gist.github.com/dhoepelman/98ffcf30e392a8f6b594).

It would be very nice if an option was added to not indent after an operator.

**Comment by ktoso** [comment]
Would indeed be nice if we could solve this (same applied to Spray back in the days, and also to the `|` operator).

I think it could be formulated as "do not indent when line ends with XXX call", perhaps having that XXX configurable?

**Comment by acjay** [comment]
Perhaps another reason to prefer https://github.com/akka/akka/issues/19678 :)

**Comment by ktoso** [comment]
yes and no. akka/akka#19678 is not a magic bullet so let's not play it as one ;-)

**Comment by acjay** [comment]
No, it's not a magic bullet. But it's possible that the hanging-indent behavior is desirable for most infix situations, but just not when the arguments themselves are really long, or multi-line, as they tend to be for `Route`. So:

```
val aggregatedResult =
  operand1 + operand2 + operand3 + operand4 +
    operand5 + operand6
```

Might make sense, but not so much:

```
val aggregatedResult = 
  bigOperation1(
    a + whole + lot + of + stuff,
    other + things) ~
  bigOperation2(
   whatever + else + is + left)
```

But one might also argue that the latter case is actually a poor fit for infix operators in the first place, as the infix operators get lost in the mix. AKA the reason for the Akka thought bubble in the first place. Right now, a missing `~` is not a compile error, but Scalariform _will_ actually help you out by formatting the cases differently, tipping you off that something is wrong. So the desire for safe code and pretty code are in conflict.

**Comment by jkinkead** [comment]
I think it's a fairly reasonable enhancement to be able to specify "non-continuation" operators in a config.

This would be a fairly big change if it were made general-purpose, since there isn't a way to provide a string (or set of strings) to a preference setting at the moment.
